pkgs/by-name/co/code-cursor/package.nix

```diff
@@ -103,36 +103,23 @@ stdenvNoCC.mkDerivation {
       #!nix-shell -i bash -p curl yq coreutils gnused trurl common-updater-scripts
       set -eu -o pipefail
 
-      baseUrl="https://download.todesktop.com/230313mzl4w4u92"
-      latestLinux="$(curl -s $baseUrl/latest-linux.yml)"
-      latestDarwin="$(curl -s $baseUrl/latest-mac.yml)"
-      linuxVersion="$(echo "$latestLinux" | yq -r .version)"
+      # Check the latest version from Cursor's website
+      version=$(curl -s "https://download.cursor.sh/latest-versions.json" | jq -r '.latestVers.main.ver')
+      buildId=$(curl -s "https://download.cursor.sh/latest-versions.json" | jq -r '.latestVers.main.buildId')
+      baseUrl="https://downloads.cursor.com/production/$buildId"
```

Comment:
You *must* check to see if the linux and darwin versions are the same before proceeding.